pkg/controller/controller.go

```diff
@@ -263,7 +263,7 @@ func (e *operationError) Error() string { return e.message }
 // The ClusterServicePlan returned will be nil if the ClusterServicePlanRef
 // is nil. This will happen when deleting a ServiceInstance that previously
 // had an update to a non-existent plan.
-func (c *controller) getClusterServiceClassPlanAndClusterServiceBroker(instance *v1beta1.ServiceInstance) (*v1beta1.ClusterServiceClass, *v1beta1.ClusterServicePlan, string, osb.Client, error) {
+func (c *controller) getClusterServiceClassPlanAndClusterServiceBroker(instance *v1beta1.ServiceInstance, fetchPlan bool) (*v1beta1.ClusterServiceClass, *v1beta1.ClusterServicePlan, string, osb.Client, error) {
```

Comment:
I think we can do better than calling `getClusterServiceClassPlanAndClusterServiceBroker` with a `fetchPlan` parameter, where it is not evident when reading from the calling side what the bool is doing. It's also strange to have a function that says in its name that it's going to get the plan but then doesn't get the plan because of a parameter.

Keep the `getClusterServiceClassPlanAndClusterServiceBroker` function signature the same. Create a `getClusterServiceClassAndClusterServiceBroker` function that can be called from `prepareDeprovisionRequest`. Have both of those get functions call this function with the `fetchPlan` parameter. Or have `getClusterServiceClassPlanAndClusterServiceBroker` call `getClusterServiceClassAndClusterServiceBroker` and then get the plan, if that won't require too much rearrangement for tests.

Reply:
Done.